File: rust/src/k01.rs
use crate::executable::Executable;

pub struct K01 {}
impl K01
{
    pub fn new() -> K01
    {
        K01 {}
    }
}

impl Executable for K01
{
    fn question_1(&self)
    {
        println!("Hello World!　ようこそRust言語の世界へ！");
    }

    fn question_2(&self)
    {
        println!("Hello World!");
        println!("ようこそ");
        println!("Rust言語の世界へ！");
    }

    fn question_3(&self)
    {
        println!("整数：{}", 12345);
        println!("実数：{}", 123.456789);
        println!("文字：{}", 'X');
        println!("文字列：{}", "ABCdef");
    }

    fn question_4(&self)
    {
        println!("              ##");
        println!("             #  #");
        println!("             #  #");
        println!("            #    #");
        println!("           #      #");
        println!("         ##        ##");
        println!("       ##            ##");
        println!("    ###                ###");
        println!(" ###       ##    ##       ###");
        println!("##        #  #  #  #        ##");
        println!("##         ##    ##         ##");
        println!(" ##     #            #     ##");
        println!("  ###     ##########     ###");
        println!("     ###              ###");
        println!("        ##############");
    }
}

File: Nim/K01.nim
import IExecutable

type K01* = ref object of IExecutable

method question1*(this: K01) =
    echo "Hello World!　ようこそ　Nim言語の世界へ！"

method question2*(this: K01) =
    echo "Hello World!"
    echo "ようこそ"
    echo "Nim言語の世界へ！"

method question3*(this: K01) =
    echo "整数：", 12345
    echo "実数：", 123.456789
    echo "文字：", 'A'
    echo "文字列：", "ABCdef"

method question4*(this: K01) =
    echo "              ##"
    echo "             #  #"
    echo "             #  #"
    echo "            #    #"
    echo "           #      #"
    echo "         ##        ##"
    echo "       ##            ##"
    echo "    ###                ###"
    echo " ###       ##    ##       ###"
    echo "##        #  #  #  #        ##"
    echo "##         ##    ##         ##"
    echo " ##     #            #     ##"
    echo "  ###     ##########     ###"
    echo "     ###              ###"
    echo "        ##############"